mtags/src/main/scala-3/scala/meta/internal/pc/SemanticTokenProvider.scala

```diff
@@ -0,0 +1,207 @@
+package scala.meta.internal.pc
+import java.{util as ju}
+
+import scala.collection.mutable.ListBuffer
+import scala.meta.internal.jdk.CollectionConverters.*
+
+import scala.meta.internal.pc.SemanticTokens.*
+import scala.meta.pc.OffsetParams
+import scala.meta.tokens.*
+import dotty.tools.dotc.ast.tpd.*
+import scala.meta.internal.mtags.MtagsEnrichments.*
+import scala.annotation.switch
+
+import org.eclipse.lsp4j.SemanticTokenModifiers
+import org.eclipse.lsp4j.SemanticTokenTypes
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.core.Symbols.NoSymbol
+import dotty.tools.dotc.core.Symbols.Symbol
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Flags
+import dotty.tools.dotc.ast.Trees.Annotated
+import dotty.tools.dotc.core.Annotations
+import dotty.tools.dotc.core.Symbols
+import scala.meta.internal.parsers.SoftKeywords
+import scala.meta.pc.VirtualFileParams
+import org.checkerframework.common.value.qual.EnumVal
+
+/**
+ *  Provides semantic tokens of file(@param params)
+ *  according to the LSP specification.
+ */
+final class SemanticTokenProvider(
+    driver: InteractiveDriver,
+    params: VirtualFileParams,
+) extends SemTokenProvider:
+  type Sym = Symbol
+  type Pos = SourcePosition
+
+  implicit val ord: Ordering[NodeInfo] = Ordering.fromLessThan((ni1, ni2) =>
+    if ni1.pos.start == ni2.pos.start then ni1.pos.end < ni2.pos.end
+    else ni1.pos.start < ni2.pos.start
+  )
+  override def getTokens(): Tokens =
+    import scala.meta.*
+    implicit val dialect = scala.meta.dialects.Scala3
+    params.text().tokenize.get
+
+  object Collector extends PcCollector[NodeInfo](driver, params):
+    override def collect(
+        parent: Option[Tree]
+    )(tree: Tree, pos: SourcePosition, symbol: Option[Symbol]): NodeInfo =
+      val sym = symbol.fold(tree.symbol)(s => s)
+      if pos.exists then NodeInfo(sym, adjust(pos)._1)
```

Comment:
We adjust for backticks using `adjust` from `PcCollector`, I think we should do the same for Scala 2